src/main/java/uk/gov/pay/directdebit/mandate/services/MandateService.java

```diff
@@ -77,36 +80,33 @@ public MandateService(DirectDebitConfig directDebitConfig,
         this.paymentProviderFactory = paymentProviderFactory;
     }
 
-    public Mandate createMandate(CreateMandateRequest createRequest, String accountExternalId) {
-        return gatewayAccountDao.findByExternalId(accountExternalId)
-                .map(gatewayAccount -> {
-                    // TODO:
-                    // when we introduce GoCardless gateway accounts to work with create mandate,
-                    // then modify appropriate mandate reference values
-                    MandateBankStatementReference mandateReference = MandateBankStatementReference.valueOf(
-                            PaymentProvider.SANDBOX.equals(gatewayAccount.getPaymentProvider()) ?
-                                    RandomStringUtils.randomAlphanumeric(18) : "gocardless-default");
-
-                    Mandate mandate = aMandate()
-                            .withGatewayAccount(gatewayAccount)
-                            .withExternalId(MandateExternalId.valueOf(RandomIdGenerator.newId()))
-                            .withMandateReference(mandateReference)
-                            .withServiceReference(createRequest.getReference())
-                            .withState(MandateState.CREATED)
-                            .withReturnUrl(createRequest.getReturnUrl())
-                            .withCreatedDate(ZonedDateTime.now(ZoneOffset.UTC))
-                            .build();
-
-                    LOGGER.info("Creating mandate external id {}", mandate.getExternalId());
-                    Long id = mandateDao.insert(mandate);
-                    mandate.setId(id);
-                    mandateStateUpdateService.mandateCreatedFor(mandate);
-                    return mandate;
-                })
-                .orElseThrow(() -> {
-                    LOGGER.error("Gateway account with id {} not found", accountExternalId);
-                    return new GatewayAccountNotFoundException(accountExternalId);
-                });
+    Mandate createMandate(CreateMandateRequest createRequest, String accountExternalId) {
+        return gatewayAccountDao.findByExternalId(accountExternalId).map(gatewayAccount -> {
+            if (gatewayAccount.getAccessToken().isEmpty()) {
+                LOGGER.error("Gateway account with id {} has no access token", accountExternalId);
+                throw new NoAccessTokenException(accountExternalId);
+            }
+            MandateBankStatementReference mandateReference = MandateBankStatementReference.valueOf(
+                    PaymentProvider.SANDBOX.equals(gatewayAccount.getPaymentProvider()) ?
+                            RandomStringUtils.randomAlphanumeric(18) : "gocardless-default");
```

Comment:
Looks like there was a TODO to set the mandate reference for gocardless accounts here. Not related to what you were doing but we should check if we need a story for this.